### DISCUSSION TITLE
Add auto-grow textarea

### DISCUSSION THREAD

**Discussion by adaboese** [cschroeter/park-ui#203] [Ideas]
I've used park ui in two projects now, and I always end up needing to use something like https://www.npmjs.com/package/react-textarea-autosize

It would be nice if the default textarea component had that option built-in.

**Comment by cschroeter** [comment]
@adaboese,

Thanks for starting this conversation. It really shows the importance of a library like ours.

I need to say that adding this particular dependency to our project isn't possible. We need something that works with all JavaScript frameworks, and adding this would make setting up Park UI way more complex.

I thought about adding the code directly to the textarea component, but that turned out to be too complicated.

But here's some good news: I found some promising talks on [Twitter](https://twitter.com/jh3yy/status/1710398436917321799) about a CSS-based autosize solution.

If you can make this work, I'd be happy to include it in Park UI. It would be a great help and really improve our project.
